Makefile

```diff
@@ -17,9 +17,14 @@ VERSION=$(shell $(PYTHON) setup.py --version 2>/dev/null)
 DESTDIR=/
 AVOCADO_DIRNAME=$(shell echo $${PWD\#\#*/})
 AVOCADO_EXTERNAL_PLUGINS=$(filter-out ../$(AVOCADO_DIRNAME), $(shell find ../ -maxdepth 1 -mindepth 1 -type d))
-AVOCADO_OPTIONAL_PLUGINS=$(shell find ./optional_plugins -maxdepth 1 -mindepth 1 -type d)
-AVOCADO_PLUGINS=$(AVOCADO_EXTERNAL_PLUGINS)
-AVOCADO_PLUGINS+=$(AVOCADO_OPTIONAL_PLUGINS)
+# List of optional plugins that have to be in setup in a giver order
+# because there may be depedencies between plugins
+AVOCADO_OPTIONAL_PLUGINS_ORDERED="./optional_plugins/runner_remote"
+# Other optional plugins found in "optional_plugins" directory
+AVOCADO_OPTIONAL_PLUGINS_OTHERS=$(shell find ./optional_plugins -maxdepth 1 -mindepth 1 -type d)
+# Unique list of optional plugins
+AVOCADO_OPTIONAL_PLUGINS=$(shell (echo "$(AVOCADO_OPTIONAL_PLUGINS_ORDERED) $(AVOCADO_OPTIONAL_PLUGINS_OTHERS)" | tr ' ' '\n' | awk '!a[$$0]++'))
```

Comment:
PS: The hatred comes from me preferring `sed` to `awk`, don't take it personally...